packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts

```diff
@@ -169,7 +169,7 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
             return;
         }
         widget.updateViewVisibility(() =>
-            part.setHidden(!this.isViewVisible(viewId))
+            part.setHidden(!this.isViewVisible(viewId), widget.parent?.node.matches(':hover'))
```

Comment:
can you explain how this expression is related to the refresh toolbar decision?

Reply:
@amiramw 
Few things to remember here:
1. plugin-view-registry catches all context key changes. I cannot know if key changed is related to widget toolbar or not - didn't think I should define a format of some kind (for key name) that will help plugin-view-registry know that a refresh toolbar is wanted.  
2. Based on #1 I therefore am doing refresh toolbar any time a context key change is caught in plugin-view-registry.ts - 
if: 
a) this.isHidden is false - meaning we are not hiding widget.
b) currently mouse in hovering over the widget (after clicking on icon mouse is staying hovering on widget)
If that is the case then refreshing toolbar.